libs/clients/stripe/client.go

```diff
@@ -0,0 +1,136 @@
+package stripe
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/brave-intl/bat-go/libs/clients"
+	appctx "github.com/brave-intl/bat-go/libs/context"
+	"github.com/google/go-querystring/query"
+)
+
+// Client abstracts over the underlying client
+type Client interface {
+	CreateOnrampSession(
+		ctx context.Context,
+		integrationMode string,
+		walletAddress string,
+		sourceCurrency string,
+		sourceExchangeAmount string,
+		destinationNetwork string,
+		destinationCurrency string,
+		supportedDestinationNetworks []string,
+	) (*OnrampSessionResponse, error)
+}
+
+// HTTPClient wraps http.Client for interacting with the Stripe server
+type HTTPClient struct {
+	client *clients.SimpleHTTPClient
+}
+
+// NewWithContext returns a new HTTPClient, retrieving the base URL from the context
+func NewWithContext(ctx context.Context) (Client, error) {
+	// get the server url from context
+	serverURL, err := appctx.GetStringFromContext(ctx, appctx.StripeOnrampServerCTXKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get StripeServer from context: %w", err)
+	}
+
+	// get the server access token from context
+	accessToken, err := appctx.GetStringFromContext(ctx, appctx.StripeOnrampSecretKeyCTXKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get StripeSecretKey from context: %w", err)
```

Comment:
Would this log our client secret then?

Reply:
No, the log here is just the error, which will be either of these two fixed strings https://github.com/brave-intl/bat-go/blob/dfb249de4e8ddb9669dc23ed3aa253bee898dfb4/libs/context/keys.go#L203-L205.